### DISCUSSION TITLE
Use system ffmpeg (linux)

### DISCUSSION THREAD

**Discussion by theofficialgman** [rustdesk/rustdesk#2800] [General]
I have recently tested the 18.04 bionic ARM64 deb from the nightly builds here. Firstly let me say great work on getting flutter on the desktop especially considering having to build it from source.

The request is to not use this library anymore for ffmpeg support: https://github.com/21pages/hwcodec#bf73e8e650abca3e004e96a245086b3647b9d84a

(edit: actually I see that the CI disables the hwcodec library arm and arm64. so I assume you are using a software encoder library not even from ffmpeg currently)

I would like for ffmpeg support to utilize the OS level ffmpeg. On some systems (such as the nintendo switch and nvidia jetson) we have hardware accelerated decoding support through custom decoders implemented and used by default in the system ffmpeg installed version.

Specifically for jetsons, we have an ffmpeg 4.4 based branch as well implementing this decoder for projects that must compile ffmpeg directly into their build (projects such a moonlight-qt use this decoder) https://gitlab.com/switchroot/switch-l4t-multimedia/FFmpeg

the codecs are called `h264_nvv4l2` and `hevc_nvv4l2` respectivly and can be enabled with `-enable-nvv4l2`. more information can be found in the commit history

**Comment by Botspot** [comment]
Raspberry Pi users would also benefit from this change. Our distro's ffmpeg has substantial optimizations and acceleration built-in.
To get a sense of the changes made, take a look at the main patch file: https://github.com/RPi-Distro/ffmpeg/blob/pios/bullseye/debian/patches/ffmpeg-4.3.4-rpi_14.patch?raw=true

**Comment by rustdesk** [comment]
PR welcome

**Reply by theofficialgman** [reply]
to be honest. your current buildsystem is an absolute nightmare. it isn't really redeamable.

if you want a PR, you first need to put in work to make hwcodec a proper library and actually buildable, instead of a git repo with binary blobs and a text file that "might" be used by a dev for building or might not be.

I am neither an ffmpeg nor rust developer but this repo looks significantly more polished than your current methods
https://github.com/angelcam/rust-ac-ffmpeg
